### PR TITLE
custom component for text, number, phonenumber

### PR DESCRIPTION
**Pull request by sudheshna-donthineni**

### CODE REVIEW

File: src/components/Datepicker.vue
Comment (sudheshna-donthineni) [inline src/components/Datepicker.vue:1]: Added a range for datepicker.

File: src/components/Dropdown.vue
Comment (sudheshna-donthineni) [inline src/components/Dropdown.vue:1]: Cleanup of unused variables.

File: src/components/EmailEntry.vue
Comment (sudheshna-donthineni) [inline src/components/EmailEntry.vue:1]: New component for email entry.

File: src/components/NumberEntry.vue
Comment (sudheshna-donthineni) [inline src/components/NumberEntry.vue:1]: Removed formkit component and built own.

File: src/components/TextEntry.vue
Comment (sudheshna-donthineni) [inline src/components/TextEntry.vue:1]: Removed formkit component and built own.

File: src/index.css
Comment (sudheshna-donthineni) [inline src/index.css:1]: I have no clue why formatting happened as of a sudden.

File: src/pages/Home.vue
Comment (sudheshna-donthineni) [inline src/pages/Home.vue:1]: No code change at all.
I have no clue why formatting happened as of a sudden.

File: src/pages/NewSignup.vue
Comment (sudheshna-donthineni) [inline src/pages/NewSignup.vue:1]: Renamed Sign in to Login and Sign up to Register.

File: src/components/NewPhoneNumberEntry.vue [hunk 1]
```diff
@@ -108,12 +113,7 @@ export default {
     isValidPhoneNumberEntry(event) {
       if (this.phoneNumber.length < 1) {
         if (
-          !(
-            event.key == "6" ||
-            event.key == "7" ||
-            event.key == "8" ||
-            event.key == "9"
-          )
+          !(event.key == "6" || event.key == "7" || event.key == "8" || event.key == "9")
```
Comment (Bahugunajii) [inline src/components/NewPhoneNumberEntry.vue:116]:
```suggestion
         if (!["6", "7", "8", "9"].includes(event.key)) {
    event.preventDefault();
}
```

Comment (Bahugunajii) [reply src/components/NewPhoneNumberEntry.vue:116]: Also maybe we can have a separate file where all these type of constants are defined

File: src/components/Datepicker.vue [hunk 2]
```diff
@@ -1,8 +1,6 @@
 <template>
   <div v-if="show">
-    <p class="text-base mb-[10px]">
-      {{ label }}<span v-if="isRequired">*</span>
-    </p>
+    <p class="text-base mb-[10px]">{{ label }}<span v-if="isRequired">*</span></p>
```
Comment (Bahugunajii) [inline src/components/Datepicker.vue:3]: We are using same class at multiple places. Let's define it once and use it everywhere.

File: src/components/Datepicker.vue [hunk 3]
```diff
@@ -18,6 +16,9 @@
         'mt-[10px] text-sm text-grey italic': true,
       }"
     />
+    <span v-if="isInvalidDateEntryMessageShown" class="text-red text-sm mt-[10px]">{{
```
Comment (Bahugunajii) [inline src/components/Datepicker.vue:19]: Same feedback as above

**Review by Bahugunajii — CHANGES_REQUESTED** [review]

File: src/components/Datepicker.vue (hunk 2 above)
Comment (sudheshna-donthineni) [reply src/components/Datepicker.vue:3]: I think they should be kept component specific. This enables for component specific formatting whenever necessary.

File: src/components/NewPhoneNumberEntry.vue (hunk 1 above)
Comment (sudheshna-donthineni) [reply src/components/NewPhoneNumberEntry.vue:116]: Again, I don't think it's necessary. Broad level validations are already in a different file. This validation is specific to the component.

**Pull request merged**